controllers/grid/users/stageParticipant/StageParticipantGridRow.inc.php

```diff
@@ -84,13 +85,36 @@ function initialize($request, $template = null) {
 			) {
 				$dispatcher = $router->getDispatcher();
 				import('lib.pkp.classes.linkAction.request.RedirectConfirmationModal');
+				$userGroupDAO = DAORegistry::getDAO('UserGroupDAO');
+				$userGroup = $userGroupDAO->getById($userGroupId, $context->getId());
+
+				if ($userGroup->getRoleId() == ROLE_ID_AUTHOR) {
+					$redirectUrl = $dispatcher->url(
+						$request,
+						ROUTE_PAGE,
+						$context->getPath(),
+						'authorDashboard',
+						'submission',
+						$submission->getId()
+					);
+				} else {
+					$redirectUrl = $dispatcher->url(
+						$request,
+						ROUTE_PAGE,
+						$context->getPath(),
+						'workflow',
+						'access',
+						$submission->getId()
+					);
+				}
```

Comment:
@forgive38 Here we could have the `$redirectUrl` be assigned bellow the `if` statement, with only the `'authorDashboard'` or `'workflow'` and `'submission'` or `'access'` assigned to two separate variables, and then passed as parameters to the `->url(..)` function.